flexport/latitude
# Component demos, meet LiveDemo

## Description:

In this PR, I mass-convert all `type: "code"` component demos under the top-level `__demo__` directory.

These components are converted to use the new `LiveDemo` component that I added to the monorepo. When viewed on the sandbox, these demos are fully editable.

I used the following rules during the conversion:
1. The converted demos should look and behave identically as the original
2. The title and descriptions of the demos are preserved (Except for newlines, which are collapsed to a single space)
3. Be as complete as possible: That is, for each demo file, I aimed to put all the code into the exported functional component. Further, I avoided importing constants from outside of the demo file. There are some downsides to this rule (We may define the same thing in multiple files), but I think the upside of showing users all of the code outweighs these negatives.

In addition to updating component demos, I also deleted the following demos:
1. `HorizontalGroup-demo`
2. `LinkAction-demo`
3. `VerticalGroup-demo`

These demos do not appear on our site, hence their source code should be deleted as well.

## Sandbox server:
https://core-5.sandbox2.eng-dev.flexport.com

## Jira:
https://flexport.atlassian.net/browse/LDS-408

## Manual Testing Plan:
Please check out the following components on the [sandbox](https://www-5.sandbox2.eng-dev.flexport.com/design/components/) and ensure that their demos all work and preserve their old purpose:

- [ ] Badge
- [ ] Banner
- [ ] Breadcrumbs
- [ ] Drawer
- [ ] DropdownButton
- [ ] Flag
- [ ] GraphicIcon
- [ ] Group
- [ ] HelpTooltip
- [ ] InlineEdit
- [ ] InputGroup
- [ ] MultiInput
- [ ] Pill
- [ ] ProgressBar
- [ ] SettingsToggle
- [ ] Text
- [ ] TextareaInput
- [ ] TextInput
- [ ] TextInputAutocomplete
- [ ] TextLink

## Screenshots:
TODO

## FYI
@flexport/theseus
